### PR TITLE
 Add MPI performance regression tests to CI performance app

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
## PR Summary

Does what it says, adds the mpi performance regression test to the performance regression apps execution cycle.

Also adds a small formatting improvement to how legends are displayed for regression performance plots on the develop branch, which you can see here has a formatting problem. https://github.com/lanl/parthenon/wiki/develop_develop

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [x] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [x] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Review by Yurlungur — APPROVED** [review]
Looks good! Where can I see the pretty pictures? :)

**Comment by JoshuaSBrown** [comment]
> Looks good! Where can I see the pretty pictures? :)

Ya, sorry something annoying happened with the ci. Hopefully it will work this time.

**Comment by JoshuaSBrown** [comment]
ping @pgrete @jlippuner

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -312,13 +312,13 @@ For the calculate pi example shown above this consists of adding parameters to f
 
 ```
 list(APPEND TEST_DIRS calculate_pi)
-list(APPEND TEST_PROCS ${NUM_MPI_PROC_TESTING})
+list(APPEND TES_PROCS ${NUM_MPI_PROC_TESTING})
```
Comment (pgrete) [inline CONTRIBUTING.md]:
```suggestion
list(APPEND TEST_PROCS ${NUM_MPI_PROC_TESTING})
```

Looks like an accidental typo (or did you intend to fix sth here?)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -312,13 +312,13 @@ For the calculate pi example shown above this consists of adding parameters to f
...
 list(APPEND TEST_ARGS "--driver ${CMAKE_BINARY_DIR}/example/calculate_pi/parthenon-example --driver_input ${CMAKE_CURRENT_SOURCE_DIR}/test_suites/calculate_pi/parthinput.regression")
 list(APPEND EXTRA_TEST_LABELS "perf")
 ```
 
 NOTE: The TEST\_PROCS list indicates how many processors to use when running
-mpi. The cmake variable NUM\_MPI\_PROC\_TESTING can be used if you do not want
+mpi. The cmake vriable NUM\_MPI\_PROC\_TESTING can be used if you do not want
```
Comment (pgrete) [inline CONTRIBUTING.md]:
```suggestion
mpi. The cmake variable NUM\_MPI\_PROC\_TESTING can be used if you do not want
```

Looks like an accidental typo (or did you intend to fix sth here?)

**Review by pgrete — APPROVED** [review]
Looks good (pending the two typos?) and I also see the pretty picture :)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -327,6 +327,17 @@ threads is by default set to 1 but can be adjusted in the driver input file
 deck. If parthenon is compiled with CUDA enabled, by default a single GPU will
 be assigned to each node..
 
+##### Integratingthe Regression Test with the Github python performance regression app
+
+In addition to running regression tests, the parthenon ci makes use of a Github python app to
```
Comment (jlippuner) [inline CONTRIBUTING.md]:
```suggestion
In addition to running regression tests, the Parthenon CI makes use of a Github python app to
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -327,6 +327,17 @@ threads is by default set to 1 but can be adjusted in the driver input file
...
+report performance metrics on machines of interest i.e. (power9 nodes). Currently, the apps source
+files are located in parthenon/scripts/python/packages/parthenon_performance_app. To add
+additional tests metrics can be plotted by making changes to these scripts. In general, the app
```
Comment (jlippuner) [inline CONTRIBUTING.md]: Incomplete sentence: "To add additional tests metrics can be plotted by making changes to these scripts."

File: scripts/python/packages/parthenon_performance_app/parthenon_performance_app/githubapp.py [hunk 5]
```diff
@@ -580,11 +580,16 @@ def cloneWikiRepo(self):
             # Ensure local branches are synchronized with server
             g.execute(['git', 'fetch'])
             # Will not overwrite files but will reset the index to match with the remote
-            g.execute(['git', 'reset','--mixed','origin/master'])
+            if wiki_state == "hard":
+              g.execute(['git', 'reset','--hard','origin/master'])
+            elif wiki_state == "mixed":
+              g.execute(['git', 'reset','--mixed','origin/master'])
+            elif wiki_state == "soft":
+              g.execute(['git', 'reset','--soft','origin/master'])
```
Comment (jlippuner) [inline scripts/python/packages/parthenon_performance_app/parthenon_performance_app/githubapp.py:588]: What if `wiki_state` is something unexpected? Should there be a default choice or error/warning message?

**Review by jlippuner — APPROVED** [review]
Looks great! I have just a few minor, non-blocking comments.

File: scripts/python/packages/parthenon_performance_app/parthenon_performance_app/githubapp.py (hunk 5 above)
Comment (JoshuaSBrown) [reply scripts/python/packages/parthenon_performance_app/parthenon_performance_app/githubapp.py:588]: That's a fair point, good catch!

File: CONTRIBUTING.md (hunk 4 above)
Comment (JoshuaSBrown) [reply CONTRIBUTING.md]: Nice catch.

**Pull request merged**